### PR TITLE
tests/utils: Move MountCloudInitFunc to local files

### PR DESCRIPTION
**Pull request by RamLavi**
### What this PR does
This PR moves MountCloudInitFunc function that is located on tests/utils to the files that use it, to be consumes locally.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
Thanks
imho the granularity of the commits can be reduced, i.e squash the two last commits at least

in this commit
https://github.com/kubevirt/kubevirt/pull/12897/commits/955125a5de826567b9076f9d85e97f725d95b5e7
worth to change the commit message to say something "move the last occurrence of the func from utils"
or even better do it agnostic to how many left and the dependency of the previous changes that reduced the count

beside that lgtm

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.

Could you please reason why it is better to have two copies of the function in question over having it in a single place?

Is there a library for cloud init under the `tests` directory?

**Comment by orelmisan** [comment]
> > Thank you for the PR @RamLavi.
> > Could you please reason why it is better to have two copies of the function in question over having it in a single place?
> 
> My best reasoning is the there is no advantage in keeping the function in a junk-pile utils file. IMO It's better to keep it close to it's use in this case.
> 
> > Is there a library for cloud init under the `tests` directory?
> 
> That could be an alternative, we have [tests/libnet/cloudinit/cloudinit.go](https://github.com/kubevirt/kubevirt/blob/main/tests/libnet/cloudinit/cloudinit.go) - but it's a strech. @orelmisan WDYT?

How about creating tests/libcloudinit? Since the other test is not a network test.
And there is at least one more function named `CheckCloudInitMetaData` that may also fit there.

**Review by dankenigsberg — CHANGES_REQUESTED** [review]

File: tests/network/sriov.go [hunk 1]
```diff
@@ -787,3 +786,18 @@ func sriovNodeName(sriovResourceName string) (string, error) {
 	}
 	return sriovNodes[0].Name, nil
 }
+
+func mountCloudInitFunc(devName string) func(*v1.VirtualMachineInstance) {
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply tests/network/sriov.go:790]: +1

**Comment by RamLavi** [comment]
> > > Thank you for the PR @RamLavi.
> > > Could you please reason why it is better to have two copies of the function in question over having it in a single place?
> > 
> > 
> > My best reasoning is the there is no advantage in keeping the function in a junk-pile utils file. IMO It's better to keep it close to it's use in this case.
> > > Is there a library for cloud init under the `tests` directory?
> > 
> > 
> > That could be an alternative, we have [tests/libnet/cloudinit/cloudinit.go](https://github.com/kubevirt/kubevirt/blob/main/tests/libnet/cloudinit/cloudinit.go) - but it's a strech. @orelmisan WDYT?
> 
> How about creating tests/libcloudinit? Since the other test is not a network test. And there is at least one more function named `CheckCloudInitMetaData` that may also fit there.

The name MountCloudInitFunc is not a good one imo. It doesn't really matter that the mount was added via CloudInit/manually/northern-elves. What it does is mounts a device to the /mnt folder. So I don't think it needs to be in a libCloudinit. 
It's just a function that runs some code on the guest terminal. And as we don't have a library for "general commands that runs on the console" (nor there should be imo) then I think it could stay on the local file where is it used.

File: tests/network/sriov.go [hunk 2]
```diff
@@ -787,3 +784,15 @@ func sriovNodeName(sriovResourceName string) (string, error) {
 	}
 	return sriovNodes[0].Name, nil
 }
+
+func mountDeviceToFS(vmi *v1.VirtualMachineInstance, devName string) error {
```
Comment (dankenigsberg) [inline tests/network/sriov.go:788]:
```suggestion
func mountGuestDevice(vmi *v1.VirtualMachineInstance, devName string) error {
```
nit: mounting is ALWAYS "to filesystem".

It's so much better without the lambda.

Comment (RamLavi) [reply tests/network/sriov.go]: I agree. [DONE](https://github.com/kubevirt/kubevirt/compare/f9b5bb28e4135a70722fd5c64a12ca6981241e20..072925a21561823f770f9fac163b69b28847c83e)

**Review by dankenigsberg — APPROVED** [review]

**Review by enp0s3 — COMMENTED** [review]
/approve
@RamLavi Thank you!

**Pull request merged**